sphinx/source/conf.py

```diff
@@ -75,6 +77,11 @@
 # Inlcude init docstrings into body of autoclass directives
 autoclass_content = "both"
 
+# Preserve signature defaults
+# Prevents entire tensors from being printed, & gives callable functions
+# proper names
+autodoc_preserve_defaults = True
```

Comment:
Can you point me to where we use variables as default? I am thinking for some of these cases, we can set the default as `None` and assign the variable within the code after checking `None`

Reply:
@aobo-y Here's an example: https://captum.ai/api/influence.html#similarityinfluence

The `similarity_metric` variable is set to the `cosine_similarity` function, and it shows up in the docs with additional characters wrapped around the name:

```
similarity_metric=<function cosine_similarity>
```

Setting `autodoc_preserve_defaults = True` corrects it.

```
similarity_metric=cosine_similarity
```


This setting is also something I need for the Optim module's TransformationRobustness class:

```
# Define TransformationRobustness defaults externally for easier Sphinx docs formatting
_TR_TRANSLATE: List[int] = [4] * 10
_TR_SCALE: List[float] = [0.995**n for n in range(-5, 80)] + [
    0.998**n for n in 2 * list(range(20, 40))
]
_TR_DEGREES: List[int] = (
    list(range(-20, 20)) + list(range(-10, 10)) + list(range(-5, 5)) + 5 * [0]
)


class TransformationRobustness(nn.Module):
    def __init__(
        self,
        padding_transform: Optional[nn.Module] = nn.ConstantPad2d(2, value=0.5),
        translate: Optional[Union[int, List[int]]] = _TR_TRANSLATE,
        scale: Optional[NumSeqOrTensorOrProbDistType] = _TR_SCALE,
        degrees: Optional[NumSeqOrTensorOrProbDistType] = _TR_DEGREES,
        final_translate: Optional[int] = 2,
        crop_or_pad_output: bool = False,
    ) -> None:
```

This way I don't need a second variable to disable each option, and Sphinx doesn't try to show hundreds of numbers in the HTML docs.

Reply:
generally, we do not encourage using variables as default. But there are surely some cases that it may make more sense to use variables.  Using huge list/tensor as default is something that does not exist before. We can discuss the optim example further when reviewing the optim module.

For existing code, I believe the only variable defaults are all functions, like the example you give. Then actually, `<function cosine_similarity>` is slightly more preferable, but no big difference, so we can keep this change. 

Please correct me if you are aware of other cases in our codebase.

Reply:
@aobo-y I think that there may be a few other variables with function defaults cases, but I don't recall any other variables in the Captum at the moment that would be affected by this change.